### PR TITLE
65 implement web api for afrr service

### LINKED ISSUE #65: Implement Web API for aFRR service
- [x] Create Bid Controller with functionality **@martin-vlacil** 
- [x] Create Signal Controller with functionality **@martin-vlacil** 
- [x] Create Dtos b
- [x] Setup DTO Converter with AutoMapper **@martin-vlacil** 
- [x] Add Dependency Injection **@martin-vlacil** 
- [x] Add Logger middleware **@WrexBG** 
- [x] Tests **@WrexBG** 
~~Add NuGet package for DataAccess in the WebAPI~~ My bad, this is needed for the other WebAPI, they are in the same solution here, ~ Alex

### PR DESCRIPTION
**Pull request by martin-vlacil**
Web API Controllers for Bids and Signals and DTOs as well as testing

### CODE REVIEW

**Comment by WrexBG** [comment]
Depends on #79 
Waiting for that to be merged
+ Doesn't have a version tag yet so that should be added before merging

File: aFRR-Service/WebAPI/Controllers/BidsController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<BidDTO>>> GetAllAsync()
+    {
+        _logger.LogInformation("GetAllAsync method called.");
+        IEnumerable<Bid> bids = await _bidDataAccess.GetAllAsync();
+        _logger.LogInformation("Finished getting {Count} bids from data access layer.", bids.Count());
+        IEnumerable<BidDTO> bidDtos = DTOConverter<Bid, BidDTO>.FromList(bids);
```
Comment (mette-thorhauge) [inline aFRR-Service/WebAPI/Controllers/BidsController.cs:47]:
```suggestion
        
        IEnumerable<BidDTO> bidDtos = DTOConverter<Bid, BidDTO>.FromList(bids);
```

File: aFRR-Service/WebAPI/Controllers/SignalsController.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<SignalDTO>>> GetAllAsync()
+    {
+        _logger.LogInformation("Getting all signals async.");
+        IEnumerable<Signal> signals = await _signalDataAccess.GetAllAsync();
+        _logger.LogInformation("Retrieved {Count} signals from the database.", signals.Count());
+        _logger.LogInformation("Converting Signal to SignalDTO.");
```
Comment (mette-thorhauge) [inline aFRR-Service/WebAPI/Controllers/SignalsController.cs:62]:
```suggestion
       
         _logger.LogInformation("Converting Signal to SignalDTO.");
```

**Review by mette-thorhauge — CHANGES_REQUESTED** [review]
The code has been commented, except for everything related to the Data access layer, as this is being changed in another project, and then implemented in this one as well. This also goes for the tests for the data access layer.

File: aFRR-Service/TestWebAPI/Tests/TestBidController.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        _webApiController = new BidsController(new BidStub(), new Mock<ILogger<BidsController>>().Object);
+    }
+
+    [Test]
+    public async Task BidsController_ShouldReturnBidDtoIdAndStatusCode200_WhenGettingValidBidId()
+    {
+        //Arrange
+        //Act
+        var actionResult = (await _webApiController.GetByIdAsync(0)).Result;
+        Assert.That(actionResult, Is.Not.Null, "Action result was null");
```
[the remark this answers is not in the document]
Comment (WrexBG) [reply aFRR-Service/TestWebAPI/Tests/TestBidController.cs]: Approved with: Commit https://github.com/diversiwaver/aFRR-NET/pull/79/commits/ede8e0aa313fbfbeebc407e8140fcf99b3f6a5be

**Review by AlexStavrev — APPROVED** [review]
Looks good

**Review by martin-vlacil — COMMENTED** [review]
gut

**Review by mette-thorhauge — APPROVED** [review]
It looks great now!

**Pull request merged**